third_party/move/move-vm/runtime/src/interpreter_caches.rs

```diff
@@ -29,15 +30,30 @@ impl Hash for FunctionPtr {
     }
 }
 
+/// Stable pointer identity for a generic [Function] within a single interpreter invocation.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub(crate) struct GenericFunctionPtr(FunctionPtr, TyArgsFingerprint);
+
+impl GenericFunctionPtr {
+    pub(crate) fn from_loaded_function(
+        function: &LoadedFunction,
+        fingerprint: TyArgsFingerprint,
+    ) -> Self {
+        Self(FunctionPtr::from_loaded_function(function), fingerprint)
+    }
+}
+
 /// Interpreter-level caches for function data (single-threaded)
 pub struct InterpreterFunctionCaches {
     function_instruction_caches: HashMap<FunctionPtr, Rc<RefCell<FrameTypeCache>>>,
```

Comment:
Would be nice to add a doc comments what those two caches do. Typically conceptual documentation at the data model is the most helpful location.

Reply:
Done